Classes/Hooks/FileDumpHook.php

```diff
@@ -199,9 +199,15 @@ protected function dumpFileContents($file, $asDownload, $resumableDownload)
             $file->getStorage()->dumpFileContents($file, $asDownload);
             exit;
         }
+        $download_name = $file->getProperty('download_name') ? $file->getProperty('download_name') : $file->getName();
+        $file_parts = pathinfo($download_name);
+        if( $file_parts['extension'] == '' ){
+        	$download_name = preg_replace( ['/\W/'], ['_'], $download_name ).'.'.$file->getExtension(); // append the original extension
```

Comment:
Why replace the spaces in the download name. Spaces are valid characters for a filename and will not give any issues. 

Would you mind removing this `preg_replace()`?